### PR TITLE
Generate negative scenarios where mandatory headers are removed

### PR DESCRIPTION
**Pull request by yogeshnikam671**
Add support for generation of negative test scenarios where mandatory headers are removed from the request.

### CODE REVIEW

File: core/src/main/kotlin/io/specmatic/core/HttpHeadersPattern.kt [hunk 1]
```diff
@@ -214,16 +214,39 @@ data class HttpHeadersPattern(
...
             patternMapR.ifValue { patternMap ->
                 HttpHeadersPattern(
                     patternMap.mapKeys { withoutOptionality(it.key) },
                     contentType = contentType
                 )
             }
+        }.plus(patternsWithNoRequiredHeaders(pattern))
+    }
+
+    private fun patternsWithNoRequiredHeaders(
```
Comment (joelrosario) [inline core/src/main/kotlin/io/specmatic/core/HttpHeadersPattern.kt:233]: This function seems identical to `patternsWithNoRequiredQueryParams`. Could they not just be the same function?

File: core/src/test/kotlin/io/specmatic/conversions/OpenApiSpecificationTest.kt [hunk 2]
```diff
@@ -8864,6 +8864,58 @@ paths:
         }
     }
 
+    @Nested
+    inner class NegativeScenariosForHeaders {
+        @Test
+        fun `should generate the negative scenarios where the mandatory headers are missing`() {
```
Comment (joelrosario) [inline core/src/test/kotlin/io/specmatic/conversions/OpenApiSpecificationTest.kt:8870]: We need one more test here showing that this functionality works when there is an example (given an example, we'd still generate a negative scenario for that example where the mandatory header is missing).

**Review by joelrosario — CHANGES_REQUESTED** [review]

File: core/src/test/kotlin/io/specmatic/conversions/OpenApiSpecificationTest.kt (hunk 2 above)
Comment (yogeshnikam671) [reply core/src/test/kotlin/io/specmatic/conversions/OpenApiSpecificationTest.kt:8870]: Addressed.

File: core/src/main/kotlin/io/specmatic/core/HttpHeadersPattern.kt (hunk 1 above)
Comment (yogeshnikam671) [reply core/src/main/kotlin/io/specmatic/core/HttpHeadersPattern.kt:233]: Addressed.

**Review by joelrosario — APPROVED** [review]

**Pull request merged**